books/spiders/books.py

```diff
@@ -0,0 +1,41 @@
+import scrapy
+from scrapy.http import Response
+
+
+class BooksSpider(scrapy.Spider):
+    name = "books"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/"]
+    rating = {
+        "One": 1,
+        "Two": 2,
+        "Three": 3,
+        "Four": 4,
+        "Five": 5
+    }
+
+    def parse_single_book(self, response: Response) -> dict:
```

Comment:
Do you need it in a class function? You can make it out-of-class, just put "rating" variable inside function